src/parser/declarations.go

```diff
@@ -310,20 +310,48 @@ func (p *parser) parseFunctionAliases(params []ast.ParameterInfo, validate func(
 		// scan the raw alias withouth the ""
 		didError := false
 		errHandleWrapper := func(err ddperror.Error) { didError = true; p.errorHandler(err) }
-		if alias, err := scanner.ScanAlias(*v, errHandleWrapper); err == nil && !didError {
-			if len(alias) < 2 { // empty strings are not allowed (we need at leas 1 token + EOF)
+
+		scanAndValidate := func(negated bool) {
+			alias, err := scanner.ScanAlias(*v, errHandleWrapper)
+			if err != nil && didError {
+				return
+			}
+
+			if len(alias) < 2 { // empty strings are not allowed (we need at least 1 token + EOF)
 				p.err(ddperror.SEM_MALFORMED_ALIAS, v.Range, "Ein Alias muss mindestens 1 Symbol enthalten")
 			} else if err := p.validateFunctionAlias(alias, params); err == nil { // check that the alias fits the function
 				if ok, isFun, existingAlias, pTokens := p.aliasExists(alias); ok {
 					p.err(ddperror.SEM_ALIAS_ALREADY_TAKEN, v.Range, ddperror.MsgAliasAlreadyExists(v.Literal, existingAlias.Decl().Name(), isFun))
 				} else {
-					funcAliases = append(funcAliases, &ast.FuncAlias{Tokens: alias, Original: *v, Func: nil, Args: paramTypesMap})
+					funcAliases = append(funcAliases, &ast.FuncAlias{Tokens: alias, Original: *v, Func: nil, Args: paramTypesMap, Negated: negated})
 					funcAliasTokens = append(funcAliasTokens, pTokens)
 				}
 			} else {
 				p.errVal(*err)
 			}
 		}
+
+		negMarkerStart := strings.Index(v.Literal, "<!")
+		if negMarkerStart != -1 {
+			if !p.isCurrentFunctionBool {
+				p.err(ddperror.SEM_ALIAS_BAD_ARGS, v.Range, "Eine Funktion die kein Wahrheitswert zurück gibt, darf auch keine Negationsmarkierungen haben")
+				continue
+			}
+
+			if strings.Contains(v.Literal[negMarkerStart+1:], "<!") {
+				p.err(ddperror.SEM_MALFORMED_ALIAS, v.Range, "Der Alias enthält mehr als eine Aliasnegationsmarkierung")
```

Comment:
Weiß nicht ob wir das als Fehler haben wollen. Natürlich macht doppelte Verneinung kein Sinn bei dem Feature, aber auch wenn ich grad kein Beispiel hab glaube ich gibt es Sätze wo man für die Verneinung mehrere Sachen ändern muss. Kommt beim Programmieren sicherlich nicht so oft vor, wäre aber vlcht nicht schlecht. Zur Not machen wir das halt sobald sich jemand beschwert.

Reply:
Ja ich würde mal das feature so shippen und abwarten. Wenn uns auffällt, dass es nützlich ist mehrere Markierungen zu haben, dann erweitern wir es dementprechend.

Ich sehe drei Wege wie man mit mehreren Markierungen umgehen kann:
1. Alle Markierungen müssen im aufruf gefüllt sein, dass der Aufruf negiert wird.
2. Eine gefüllte Markierung toggelt, den Negierungsstatus.
3. Mehrere Markierungen erzeugen jede Kombination von Aliasen, welche eine Markierung enthalten